controllers/nutanixcluster_controller.go

```diff
@@ -170,20 +170,11 @@ func (r *NutanixClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		log.V(1).Info(fmt.Sprintf("Patched NutanixCluster. Status: %+v", cluster.Status))
 	}()
 
-	err = r.reconcileCredentialRef(ctx, cluster)
```

Comment:
What do you think about keeping `r.reconcileCredentialRef` but also returning the `Secret` and then passing that Secret into `getPrismCentralClientForCluster`, that would make the function a little more focused and more unit testable?

It would also make the changes in this PR smaller

Reply:
I've narrowed down the focus of this PR to just enabling session-auth. I've moved the client caching scope to another PR. https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/pull/415